### PR TITLE
Redeem widget

### PR DESCRIPTION
**Pull request by memosys**
Contains Working `Redemption Widget`

-  UX could be improved
-  Made some null/default value handling improvements on swap widget as well
- switched to using form on swap/redeem widgets

### CODE REVIEW

File: web-portal/backend/src/utils/utils.service.ts [hunk 1]
```diff
@@ -151,8 +151,7 @@ export class UtilsService {
         });
 
         const parsedLogs: IParsedLog[] = logs.map((log: any) => ({
-            tenantId: web3.utils
-                .toAscii(log?.args?._identifier!)
+            tenantId: fromHex(log?.args?._identifier!, 'string')
                 .replaceAll(`\x00`, ''),
```
Comment (plor) [inline web-portal/backend/src/utils/utils.service.ts:155]: One thing I thought of here is what happens if there are zeros in the identifier that get removed?

**Review by plor — APPROVED** [review]
Looks good, one note though, it is difficult to do reviews when large formatting changes are included. If its possible to keep them separate that would benefit the reviews.

**Comment by memosys** [comment]
> Looks good, one note though, it is difficult to do reviews when large formatting changes are included. If its possible to keep them separate that would benefit the reviews.

sorry about that, my editor is acting weird!

File: web-portal/backend/src/utils/utils.service.ts (hunk 1 above)
Comment (memosys) [reply web-portal/backend/src/utils/utils.service.ts:155]: doesn't work that way for un-padded zeroes! (hopefully)

**Pull request merged**